packages/playwright-core/src/cli/docker.ts

```diff
@@ -0,0 +1,386 @@
+ * limitations under the License.
+ */
+/* eslint-disable no-console */
+
+import http from 'http';
+import * as utils from '../utils';
+import { spawnAsync } from '../utils/spawnAsync';
+import { getPlaywrightVersion } from '../common/userAgent';
+
+interface DockerImage {
+  Containers: number;
+  Created: number;
+  Id: string;
+  Labels: null | Record<string, string>;
+  ParentId: string;
+  RepoDigests: null | string[];
+  RepoTags: null | string[];
+  SharedSize: number;
+  Size: number;
+  VirtualSize: number;
+}
+
+const VRT_IMAGE_DISTRO = 'jammy';
+const VRT_IMAGE_NAME = `mcr.microsoft.com/playwright:v${getPlaywrightVersion()}-${VRT_IMAGE_DISTRO}-vrt`;
+const VRT_CONTAINER_NAME = `playwright-${getPlaywrightVersion()}-${VRT_IMAGE_DISTRO}-vrt`;
+
+const GENERATE_FLUXBOX_BROWSERS_MENU = `
+  const { chromium, firefox, webkit } = require('playwright-core');
+
+  console.log(\`
+    [begin] (fluxbox)
+      [submenu] (Browsers) {}
+        [exec] (Chromium) { $\{chromium.executablePath()} --no-sandbox --test-type= } <>
+        [exec] (Firefox) { $\{firefox.executablePath()} } <>
+        [exec] (WebKit) { $\{webkit.executablePath()} } <>
+      [end]
+      [include] (/etc/X11/fluxbox/fluxbox-menu)
+    [end]
+  \`);
+`;
+
+const CONTAINER_ENTRY_POINT = `#!/bin/bash
+  set -e
+  SCREEN_WIDTH=1360
+  SCREEN_HEIGHT=1020
+  SCREEN_DEPTH=24
+  SCREEN_DPI=96
+  GEOMETRY="$SCREEN_WIDTH""x""$SCREEN_HEIGHT""x""$SCREEN_DEPTH"
+
+  nohup /usr/bin/xvfb-run --server-num=$DISPLAY_NUM \
+       --listen-tcp \
+       --server-args="-screen 0 "$GEOMETRY" -fbdir /var/tmp -dpi "$SCREEN_DPI" -listen tcp -noreset -ac +extension RANDR" \
+       /usr/bin/fluxbox -display "$DISPLAY" >/dev/null 2>&1 &
+
+  for i in $(seq 1 50)
+    do
+      if xdpyinfo -display $DISPLAY >/dev/null 2>&1; then
+        break
+      fi
+      echo "Waiting for Xvfb..."
+      sleep 0.2
+    done
+
+
+  nohup x11vnc -forever -shared -rfbport 5900 -rfbportv6 5900 -display "$DISPLAY" >/dev/null 2>&1 &
+  nohup /opt/bin/noVNC/utils/novnc_proxy --listen 7900 --vnc localhost:5900 >/dev/null 2>&1 &
+  cd /ms-playwright-agent
+  NOVNC_UUID=$(cat /proc/sys/kernel/random/uuid)
+  echo "novnc is listening on http://127.0.0.1:7900?path=$NOVNC_UUID&resize=scale"
+  PW_UUID=$(cat /proc/sys/kernel/random/uuid)
+  npx playwright run-server --port=5400 --path=/$PW_UUID
+`;
+
+const NOVNC_REF = '1.3.0';
+const WEBSOCKIFY_REF = '0.10.0';
+const CONTAINER_BUILD_SCRIPT = `
+  # Generate entry point script
+  cat <<'EOF' >/start.sh
+  ${CONTAINER_ENTRY_POINT}
+  EOF
+  chmod 755 /start.sh
+
+  export DEBIAN_FRONTEND=noninteractive
+
+  # Install FluxBox, VNC & noVNC
+  mkdir -p /opt/bin && chmod +x /dev/shm \
+      && apt-get update && apt-get install -y unzip fluxbox x11vnc \
+      && curl -L -o noVNC.zip "https://github.com/novnc/noVNC/archive/v${NOVNC_REF}.zip" \
+      && unzip -x noVNC.zip \
+      && mv noVNC-${NOVNC_REF} /opt/bin/noVNC \
```

Comment:
```suggestion
      && rm -rf noVNC-${NOVNC_REF}/{docs,tests} \
      && mv noVNC-${NOVNC_REF} /opt/bin/noVNC \
```

not sure if we should!

Reply:
Sure why not!